src/qibolab/platforms/native.py

```diff
@@ -0,0 +1,188 @@
+
+@dataclass
+class VirtualPulse:
+    """Container with parameters required to add a virtual Z phase in a pulse sequence."""
+
+    phase: float
+    qubit: "Qubit"
+    type: str = "virtual_z"
+
+
+@dataclass
+class NativeSequence:
+    """List of :class:`qibolab.platforms.native.NativePulse` objects implementing a gate.
+
+    Relevant for two-qubit gates, which usually require a sequence of pulses to be implemented.
+    These pulses may act on qubits different than the qubits the gate is targeting.
+    """
+
+    name: str
+    pulses: List[Union[NativePulse, VirtualPulse]] = field(default_factory=list)
+
+    @classmethod
+    def from_dict(cls, name, qubits, sequence):
+        """Constructs the native sequence from the dictionaries provided in the runcard.
+
+        Args:
+            name (str): Name of the gate the sequence is applying.
+            qubits (list): List of :class:`qibolab.platforms.abstract.Qubit` object for all
+                qubits in the platform. All qubits are required because the sequence may be
+                acting on qubits that the implemented gate is not targeting.
+            sequence (dict): Dictionary describing the sequence as provided in the runcard.
+        """
+        pulses = []
+
+        # If sequence contains only one pulse dictionary, convert it into a list that can be iterated below
+        if isinstance(sequence, dict):
+            sequence = [sequence]
+
+        for i, pulse in enumerate(sequence):
+            pulse_type = pulse["type"]
+            kwargs = dict(pulse)
+            qubit = qubits[kwargs.pop("qubit")]
+            if pulse_type == "virtual_z":
+                phase = kwargs["phase"]
+                pulses.append(VirtualPulse(phase, qubit))
+            elif pulse_type == "qf":
+                name = f"{name}{i}"
+                pulses.append(NativeFluxPulse.from_dict(name, **kwargs, qubit=qubit))
+            elif pulse_type == "qd":
+                name = f"{name}{i}"
+                pulses.append(NativePulse.from_dict(name, **kwargs, qubit=qubit))
+            else:
+                raise_error(TypeError, f"Incorrect type {pulse_type} was given in NativeSequence.")
+
+        return cls(name, pulses)
+
+    def sequence(self, start=0):
+        """Creates a :class:`qibolab.pulses.PulseSequence` object implementing the sequence."""
+        sequence = PulseSequence()
+        virtual_z_phases = defaultdict(int)
+
+        for pulse in self.pulses:
+            if isinstance(pulse, NativePulse):
+                sequence.add(pulse.pulse(start=start))
+            else:
+                virtual_z_phases[pulse.qubit.name] += pulse.phase
+
+        # TODO: Maybe ``virtual_z_phases`` should be an attribute of ``PulseSequence``
+        return sequence, virtual_z_phases
+
+
+@dataclass
+class NativeSingleQubitGates:
+    """Container with the native single qubit gates acting on a specific qubit."""
+
+    MZ: NativePulse
+    RX: NativePulse
+    RX90: NativePulse
```

Comment:
I know it looks nicer like this, but in principle it would be better to follow usual naming conventions:
```suggestion
    mz: NativePulse
    rx: NativePulse
    rx90: NativePulse
```

Translating from one to the other is an automated process anyhow (in case you get them as strings, you just need to call `.lower()` and `.upper()`, but in principle you should not get them as strings, most of the time at least).

Reply:
The main reason I prefer capitals here is agreement with qibo gates. For example the `RX` pulse here is basically the hardware equivalent of the `RX(theta=np.pi)` qibo gate.

Reply:
I imagined, so we should take into account if we really want to actively break conventions, or if we want to change even in Qibo.

In case of doubts, it is fine to keep as it is.